### PR TITLE
variable: add support for last_insert_id

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Problem Summary:

In MySQL, `last_insert_id` is available as either a func or a sysvar. However, in TiDB it only worked correctly as a func.

This fixes the sysvar usage.

### What is changed and how it works?

What's Changed:

The value returned by `SELECT @@last_insert_id` is now compatible with MySQL.

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note 

```release-note
The value returned by `SELECT @@last_insert_id` is now compatible with MySQL.
```

### CODE REVIEW

**Review by dveeden — APPROVED** [review]
Looks fine:
```
mysql> \W
Show warnings enabled.
mysql> CREATE TABLE t(id bigint unsigned auto_random primary key);
Query OK, 0 rows affected, 1 warning (0.01 sec)

Note (Code 1105): Available implicit allocation times: 576460752303423487
mysql> INSERT INTO t VALUES();
Query OK, 1 row affected (0.00 sec)

mysql> SELECT @@last_insert_id;
+----------------------+
| @@last_insert_id     |
+----------------------+
| 16140901064495857665 |
+----------------------+
1 row in set (0.01 sec)

mysql> TABLE t;
+----------------------+
| id                   |
+----------------------+
| 16140901064495857665 |
+----------------------+
1 row in set (0.00 sec)
```

File: expression/integration_test.go [hunk 1]
```diff
@@ -10302,3 +10302,17 @@ func (s *testIntegrationSuite) TestIssue27610(c *C) {
...
 }
+
+func (s *testIntegrationSuite) TestLastInsertId(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`use test;`)
+	tk.MustExec(`drop table if exists lastinsertid;`)
+	tk.MustExec(`create table lastinsertid (id int not null primary key auto_increment);`)
+	tk.MustQuery("SELECT @@last_insert_id;").Check(testkit.Rows("0"))
+	tk.MustExec(`INSERT INTO lastinsertid VALUES (NULL);`)
+	tk.MustQuery("SELECT @@last_insert_id, LAST_INSERT_ID()").Check(testkit.Rows("1 1"))
```
Comment (djshow832) [inline expression/integration_test.go:10313]: `LAST_INSERT_ID` can also take an expression as an argument, https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_last-insert-id
Will you support that?

Comment (djshow832) [reply expression/integration_test.go:10313]: Sorry, I misunderstood the goal of this PR.

**Review by djshow832 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**